core/src/banking_stage.rs

```diff
@@ -2003,7 +1984,12 @@ impl BankingStage {
             *newly_buffered_packets_count += packet_indexes.len();
             slot_metrics_tracker
                 .increment_newly_buffered_packets_count(packet_indexes.len() as u64);
-            unprocessed_packet_batches.push_back((packet_batch, packet_indexes, false));
+
+            unprocessed_packet_batches.push_back(DeserializedPacketBatch::new(
+                packet_batch,
+                packet_indexes,
+                false,
+            ));
```

Comment:
This change will now deserialize every `PacketBatch` before adding it to the buffered queue, whereas the current code only deserializes when we're about to process the packet.

How much of a perf hit does this incur if we're ingesting up to max buffered packets? Before we were just adding to a vector so it probably wouldn't block the transaction processing.

If the extra perf hit is significant, should we add a max time spent receiving now/max packets receive to ensure we don't spin in here for too long? https://github.com/solana-labs/solana/blob/caf2b94bb1f40d09c93a58617cd53fc621f35175/core/src/banking_stage.rs#L1894-L1915

Reply:
> mean tps went from ~20K -> ~16K

hmmm that's quite a big perf hit, that's just from this PR or the bigger PR here? https://github.com/solana-labs/solana/pull/23257

I would prefer to not check in PR's that induce a perf hit, and then try to check in fixes in later PR's. Keeping each PR perf-neutral is easier for management so that if we roll back certain PR's, we don't have to remember the interdependencies.

Here, maybe a lazy deserialization might help? Or what if we moved the deserialization into a separate stage/part of sigverify that streamed the deserialized transactions to banking stage?

Reply:
> that's just from this PR or the bigger PR here: #23257

That's from the bigger all-in-one PR. 

But this is a fair point. I just kicked a 1-hour test. Let's see the results. Definitely signup on the idea.

Reply:
cool, thanks!